builders/interoperability/xcm/xc-registration/assets.md

```diff
@@ -39,193 +39,107 @@ To create a forum post on the [Moonbeam Community Forum](https://forum.moonbeam.
+To submit a preimage, you'll need to get the encoded calldata for each extrinsic that you want to execute. As previously mentioned, you'll use the `evmForeignAssets.createForeignAsset` and the `xcmWeightTrader.addAsset` extrinsics.
 
-You can use the [`xcm-asset-registrator.ts` script](https://github.com/Moonsong-Labs/xcm-tools/blob/main/scripts/xcm-asset-registrator.ts){target=\_blank} to calculate the encoded calldata and even submit the preimage and proposal if you desire. Proposals must be submitted via the General Admin Track. If you're registering an asset and opening a channel, you'll want to wait to submit the preimage and proposal until you have the calldata for the channel-related calls.
+Proposals must be submitted via the `FastGeneralAdmin` track. If you're opening a channel and registering an asset and you'll want to wait until the channel is established prior to attempting to register the asset. 
 
-To get the encoded calldata for the `assetManager.registerForeignAsset` extrinsic, you can use the following arguments:
+To get the encoded calldata for the `evmForeignAssets.createForeignAsset` extrinsic, you will need to provide the following arguments:
 
-- `--ws-provider` or `--w` - the WebSocket provider to be used for the requests. The WSS network endpoints for each Moonbeam-based network are as follows:
+- `assetId` - unique identifier of the asset
+- `xcmLocation` - the multilocation of the asset relative to Moonbeam 
+- `decimals` - the number of decimals of the asset
+- `symbol`  - the symbol of the asset. **Remember that "xc" should be prepended to the symbol to indicate the asset is an XCM-enabled asset**
+- `name` - The asset name
 
-    === "Moonbeam"
+You can use the script below to generate the encoded call data for the `createForeignAsset` call. Remember to replace the example parameter values shown in the script with the ones relevant to your asset. 
 
-        ```text
-        wss://wss.api.moonbeam.network
-        ```
+???+ code "Get encoded call data for createForeignAsset"
+    ```typescript
+    --8<-- 'code/builders/interoperability/xcm/xc-registration/assets/create-foreign-asset.js'
+    ```
 
-    === "Moonriver"
+The resulting output will be something like:
 
-        ```text
-        wss://wss.api.moonriver.moonbeam.network
-        ```
+--8<-- 'code/builders/interoperability/xcm/xc-registration/assets/terminal/createForeignAsset.md'
 
-    === "Moonbase Alpha"
+To get the encoded calldata for the `xcmWeightTrader.addAsset` extrinsic, you will need to provide the following arguments:
 
-        ```text
-        {{ networks.moonbase.wss_url }}
-        ```
+- `xcmLocation` - the multilocation of the asset relative to Moonbeam 
+- `relativePrice` - the cost of an asset in terms of weight, used to determine how much of the asset is required to cover the fees for cross-chain (XCM) operations. It is calculated by comparing the asset's value to the network's native token in terms of the weight-to-fee conversion
```

Comment:
How do you get this? We need to provide a practical example or make `xcm-tools` calculate this for you!

Reply:
Added a script on XCM-tools :)